lib/node_modules/@stdlib/repl/lib/completer_engine.js

```diff
@@ -0,0 +1,310 @@
+			return;
+		}
+		self._completionsList = completions[ 0 ].filter( isNotEmptyString ); // remove empty completions
+		if ( self._completionsList.length === 0 ) {
+			debug( 'No completions to display.' );
+			return;
+		}
+		// Resolve a common prefix from the completion results:
+		autoCompletion = commonPrefix( self._completionsList ); // e.g., [ 'back', 'background', 'backward' ] => 'back'
+
+		// If the completion candidates have a possible auto-completion (ie. a common prefix longer than the input), auto-complete it...
+		if ( autoCompletion !== '' && autoCompletion.length > completions[ 1 ].length ) {
+			debug( 'Found an auto-completion candidate: %s', autoCompletion );
+
+			// Clear the input line
+			self._ostream.write( repeat( '\x08', completions[1].length ) );
+			self._rli.line = self._rli.line.slice( self._rli.cursor );
+
+			// Move the cursor to the start of the line
+			self._rli.cursor = 0;
+
+			// Write the auto-completion string
+			self._rli.write( autoCompletion );
+
+			return;
+		}
+		debug( 'No auto-completion candidate, displaying all possible completions.' );
+
+		// Display completions:
+		self._displayCompletions();
+
+		// Re-display the prompt:
+		displayPrompt( self.repl, true );
+	}
+});
+
+/**
+* Highlights the matching parts of the completions based on the current line.
+*
+* @private
+* @name _highlightCompletions
+* @memberof CompleterEngine.prototype
+* @returns {Array<string>} array of highlighted completions
+*/
+setNonEnumerableReadOnly( CompleterEngine.prototype, '_highlightCompletions', function highlightCompletions() {
+	var self = this;
+
+	return this._completionsList.map( function highlightCompletion( completion ) { // eslint-disable-line max-len
+		var lastMatchedIndex = -1;
+		var completionIndex = 0;
+		var highlighted = '';
+		var boldIndexes = []; // Buffer to store indexes of characters in completion string that needs to be highlighted
+		var lineIndex = 0;
+		var i;
+
+		// If input is an exact prefix of completion, directly highlight the substring...
+		if ( startsWith( completion, self._line ) ) {
+			return '\x1b[1m' + completion.slice(0, self._line.length) + '\x1b[0m' + completion.slice(self._line.length);
+		}
+
+		// Store indexes of each matching character in the completion string in the buffer...
+		while ( lineIndex < self._line.length && completionIndex < completion.length ) { // eslint-disable-line max-len
+			if ( lowercase( completion[ completionIndex ] ) === lowercase( self._line[ lineIndex ] ) ) { // eslint-disable-line max-len
+				boldIndexes.push(completionIndex);
+				lastMatchedIndex = completionIndex;
+				lineIndex += 1;
+			} else if ( completionIndex + 1 === completion.length ) {
+				lineIndex += 1;
+				completionIndex = lastMatchedIndex + 1;
+			}
+			completionIndex += 1;
+		}
+
+		// Highlight stored indexes in the completion string:
+		for ( i = 0; i < completion.length; i++ ) {
+			if ( boldIndexes.includes( i ) ) {
```

Comment:
Do not use `includes` due to backward compatibility. You can use `@stdlib/array/base/assert/contains`.